net/netmon/netmon.go

```diff
@@ -401,13 +398,23 @@ func (m *Monitor) handlePotentialChange(newState *State, forceCallbacks bool) {
 		return
 	}
 
-	delta := &ChangeDelta{
-		Monitor:    m,
-		Old:        oldState,
-		New:        newState,
+	delta := ChangeDelta{
 		TimeJumped: timeJumped,
 	}
 
+	// The subscribers can listen for nil fields, so we have to give it nillable
+	// items in the struct. Copy them so subscribers cannot change anything at the
+	// source.
+	if oldState != nil {
+		delta.Old = new(State)
```

Comment:
This is a shallow copy, which is OK, but if the goal is to avert changes on the receiving side, we might need to clone the maps too. Maybe that isn't necessary? But if not then perhaps we can share the pointers too.

If we're concerned about mutability, we might add a Clone method to the State type, though I'm not sure if that's necessary.